test/e2e/cluster_upgrade.go

```diff
@@ -170,191 +257,77 @@ func nodeUpgradeGKE(v string) error {
-		//  - secrets
-		//  - volumes
-		//  - persistent volumes
-	})
-
-	AfterEach(func() {
-		w.Cleanup()
-	})
-
-	framework.KubeDescribe("master upgrade", func() {
-		It("should maintain responsive services [Feature:MasterUpgrade]", func() {
-			By("Validating cluster before master upgrade")
-			framework.ExpectNoError(validate(f, svcName, rcName, ingress, replicas))
-			By("Performing a master upgrade")
-			testUpgrade(ip, v, masterUpgrade)
-			By("Checking master version")
-			framework.ExpectNoError(checkMasterVersion(f.Client, v))
-			By("Validating cluster after master upgrade")
-			framework.ExpectNoError(validate(f, svcName, rcName, ingress, replicas))
-		})
-	})
+func testServiceUpBeforeAndAfter(f *framework.Framework, sem *chaosmonkey.Semaphore) {
+	testService(f, sem, false)
+}
 
-	framework.KubeDescribe("node upgrade", func() {
-		It("should maintain a functioning cluster [Feature:NodeUpgrade]", func() {
-			By("Validating cluster before node upgrade")
-			framework.ExpectNoError(validate(f, svcName, rcName, ingress, replicas))
-			By("Performing a node upgrade")
-			// Circumnavigate testUpgrade, since services don't necessarily stay up.
-			framework.Logf("Starting upgrade")
-			framework.ExpectNoError(nodeUpgrade(f, replicas, v))
-			framework.Logf("Upgrade complete")
-			By("Checking node versions")
-			framework.ExpectNoError(checkNodesVersions(f.Client, v))
-			By("Validating cluster after node upgrade")
-			framework.ExpectNoError(validate(f, svcName, rcName, ingress, replicas))
-		})
+func testServiceRemainsUp(f *framework.Framework, sem *chaosmonkey.Semaphore) {
+	testService(f, sem, true)
+}
 
-		It("should maintain responsive services [Feature:ExperimentalNodeUpgrade]", func() {
-			By("Validating cluster before node upgrade")
-			framework.ExpectNoError(validate(f, svcName, rcName, ingress, replicas))
-			By("Performing a node upgrade")
-			testUpgrade(ip, v, func(v string) error {
-				return nodeUpgrade(f, replicas, v)
-			})
-			By("Checking node versions")
-			framework.ExpectNoError(checkNodesVersions(f.Client, v))
-			By("Validating cluster after node upgrade")
-			framework.ExpectNoError(validate(f, svcName, rcName, ingress, replicas))
-		})
+// testService is a helper for testServiceUpBeforeAndAfter and testServiceRemainsUp with a flag for testDuringUpgrade
+//
+// TODO(ihmccreery) remove this abstraction once testServiceUpBeforeAndAfter is no longer needed, because node upgrades
+// maintain a responsive service.
+func testService(f *framework.Framework, sem *chaosmonkey.Semaphore, testDuringUpgrade bool) {
+	// Setup
+	serviceName := "service-test"
+
+	jig := NewServiceTestJig(f.Client, serviceName)
+	// nodeIP := pickNodeIP(jig.Client) // for later
+
+	By("creating a TCP service " + serviceName + " with type=LoadBalancer in namespace " + f.Namespace.Name)
+	// TODO it's weird that we have to do this and then wait WaitForLoadBalancer which changes
+	// tcpService.
```

Comment:
The initial create does not have the LB status block.  If you don't need the LB status (e.g. IP) then you don't have to wait.  I opted to not make the create also wait because it wasn't clear how long was "long enough" in all cases, and they seemed like distinct ideas